articles/sentinel/notebooks.md

```diff
@@ -42,47 +42,123 @@ We've also released some open-source Jupyter security tools in a package named [
+    - [PowerShell](../role-based-access-control/role-assignments-powershell.md)
+    - [Azure CLI](../role-based-access-control/role-assignments-cli.md)
+    - [REST API](../role-based-access-control/role-assignments-rest.md)
+    - [Azure Resource Manager templates](../role-based-access-control/role-assignments-template.md)
 
+    If you have installed the [Azure Machine Learning CLI](https://docs.microsoft.com/azure/machine-learning/reference-azure-machine-learning-cli), you can use CLI commands to assign roles to users:
+
+    ```azurecli-interactive 
+    az ml workspace share -w <workspace_name> -g <resource_group_name> --role <role_name> --user <user_corp_email_address>
+    ```
+
+    The `user` field is the email address of an existing user in the instance of Azure Active Directory where the workspace parent subscription lives. Here is an example of how to use this command:
+
+    ```azurecli-interactive 
+    az ml workspace share -w my_workspace -g my_resource_group --role Contributor --user jdoe@contoson.com
+    ```
+
+    > [!NOTE]
+    > "az ml workspace share" command does not work for federated account by Azure Active Directory B2B. Please use Azure UI portal instead of command.
+
+    ### Grant access via Azure portal UI
+    To grant permissions via the UI, follow the following steps.
+
+    i. From the Azure portal, navigate to your Machine Learning workspace. If you don't have an Azure ML workspace, [create a new Azure ML workspace](#Create-an-Azure-ML-workspace).
+
+    ii. Select **Access Control (IAM)** blade, then in **Grant access to this resource**, select **Add Role Assignment** or **Use the classic experience**.
+
+    iii. Select a role and the user you want to grant the permissions to. Then select **Save**.
+    
     > [!div class="mx-imgBorder"]
-    > ![launch notebook to start azure ml workspace](./media/notebooks/sentinel-notebooks-launch.png)
+    > ![grant aml access](./media/notebooks/sentinel-notebooks-aml-access.png)
+
+    ### Create custom role
+    
+    If the built-in roles are insufficient, you can create custom roles. Custom roles might have read, write, delete, and compute resource permissions in that workspace. You can make the role available at a specific workspace level, a specific resource group level, or a specific subscription level.
+
+    Learn more on how to [create custom role](https://docs.microsoft.com/azure/machine-learning/how-to-assign-roles#create-custom-role).
 
-1. Under **AzureML Workspace**, select **Create New**.
+## Create an Azure ML workspace
+
+1. From the Azure portal, navigate to **Azure Sentinel** > **Threat management** > **Notebooks** and then select **Create a new AML workspace**.
 
     > [!div class="mx-imgBorder"]
-    > ![create workspace](./media/notebooks/sentinel-notebooks-azureml-create.png)
+    > ![create a new aml workspace](./media/notebooks/sentinel-azure-notebooks-aml-create.png)
```

Comment:
This image should include the bar at the top (just above the number of notebook templates) where the "Create new AML workspace" button appears.